File: apiclient/types/project.go
package types

type Project struct {
	Metadata
	ProjectManifest
	AssistantID string `json:"assistantID,omitempty"`
	Editor      bool   `json:"editor"`
	ParentID    string `json:"parentID,omitempty`
}

type ProjectManifest struct {
	ThreadManifest
}

type ProjectList List[Project]

File: ui/admin/app/lib/model/project.ts
import { EntityMeta } from "~/lib/model/primitives";
import { ThreadManifest } from "~/lib/model/threads";

type ProjectManifest = ThreadManifest & {
	parentID: string;
	assistantID: string;
	editor: boolean;
};

export type Project = EntityMeta & ProjectManifest;
